### PR TITLE
backupccl: fix scheduled backup pts pushing race

### PR DESCRIPTION
**Pull request by msbutler**
Informs #128013

Epic: none

### CODE REVIEW

File: pkg/ccl/backupccl/backup_job.go [hunk 1]
```diff
@@ -663,6 +663,21 @@ func (b *backupResumer) Resume(ctx context.Context, execCtx interface{}) error {
...
 	}
 
+	// Choose which scheduled backup pts we will update at the the end of the
+	// backup _before_ we resolve the destination of the backup. This avoids a
+	// race with inc backups where backup destination resolution leads this backup
+	// to extend a chain that is about to be superseded by a new full backup
+	// chain, which could cause this inc to accidentally push the pts for the
+	// _new_ chain instead of the old chain it is apart of. By choosing the pts to
+	// move before we resolve the destination, we guarantee that we push the old
+	// chain.
```
Comment (kev-cao) [inline pkg/ccl/backupccl/backup_job.go:673]: Isn't the destination resolved at L621?

Comment (msbutler) [reply pkg/ccl/backupccl/backup_job.go]: lol, whoops

**Review by kev-cao — APPROVED** [review]
LGTM!

**Review by dt — APPROVED** [review]

**Comment by msbutler** [comment]
TFTRs!

bors r+

**Pull request merged**